### PR TITLE
Merge `main` into `feat/multichain`

### PR DESCRIPTION
**Pull request by Wodann**
This PR merges `main` into `feat/multichain`. As such, most commits are not very interesting.

However, special care should be given to this [commit](https://github.com/NomicFoundation/edr/pull/671/commits/c68d3ab27d6528285a9e529afa7fe5c547b40634) as it reimplements the behaviour for skipping unrecognised transactions using the multi-chain architecture.

### CODE REVIEW

File: crates/edr_eth/src/transaction.rs [hunk 1]
```diff
@@ -234,6 +234,11 @@ pub trait IsEip4844 {
     fn is_eip4844(&self) -> bool;
 }
 
+pub trait IsExecutable {
+    /// Whether the transaction is executable.
+    fn is_executable_transaction(&self) -> bool;
```
Comment (Xanewok) [inline crates/edr_eth/src/transaction.rs:239]: nit: The call sites seem to assume that "executable" specifically means unsupported. Maybe we should rename it to `is(_transaction)_supported` or similar? I had to look at the call sites to understand why a transaction might not be executable, so maybe referring to the supported/recognized quality directly here would make it more obvious.

File: crates/edr_eth/src/transaction.rs [hunk 2]
```diff
@@ -234,6 +234,11 @@ pub trait IsEip4844 {
     fn is_eip4844(&self) -> bool;
 }
 
+pub trait IsExecutable {
```
Comment (Xanewok) [inline crates/edr_eth/src/transaction.rs:237]: Maybe we could add more clarification here for the trait doc comment itself

**Review by Xanewok — APPROVED** [review]
I looked specifically at the commit you linked, looks good! Left a comment but not blocking.

File: crates/edr_provider/src/data.rs [hunk 3]
```diff
@@ -2399,6 +2400,39 @@ where
...
+    fn filter_unsupported_transaction_types(
+        &self,
+        transactions: &[ChainSpecT::Transaction],
+        transaction_hash: &B256,
+    ) -> Result<Vec<ChainSpecT::Transaction>, ProviderError<ChainSpecT>> {
+        transactions
+            .iter()
+            .filter_map(|transaction| {
+                if transaction.is_executable_transaction() {
+                    Some(Ok(transaction.clone()))
```
Comment (agostbiro) [inline crates/edr_provider/src/data.rs:2416]: Cloning here is a slight performance regression compared to main, but I don't think it matters compared to the cost of executing the transactions.

**Review by agostbiro — APPROVED** [review]
Thanks, LGTM overall

File: crates/edr_provider/src/data.rs (hunk 3 above)
Comment (Wodann) [reply crates/edr_provider/src/data.rs:2416]: On main we were [calling `to_vec`](https://github.com/NomicFoundation/edr/blob/main/crates/edr_provider/src/data.rs#L1985) which would also have cloned the transactions.

https://doc.rust-lang.org/std/primitive.slice.html#method.to_vec

Comment (agostbiro) [reply crates/edr_provider/src/data.rs:2416]: Only for locally mined blocks which will typically have a lot fewer transactions than remote blocks. The remote path [takes](https://github.com/NomicFoundation/edr/blob/c8fbf3b2a6ca4f838d408a47536668a0c413c256/crates/edr_provider/src/data.rs#L2015) the transactions from the remote block.

Comment (Wodann) [reply crates/edr_provider/src/data.rs:2416]: Gotcha! 👍

File: crates/edr_eth/src/transaction.rs (hunk 1 above)
Comment (Wodann) [reply crates/edr_eth/src/transaction.rs]: Addressed in https://github.com/NomicFoundation/edr/pull/671/commits/9d3adad2a37095385ca9a6f7ff5c9c719e842f2c

File: crates/edr_eth/src/transaction.rs (hunk 2 above)
Comment (Wodann) [reply crates/edr_eth/src/transaction.rs]: Addressed in https://github.com/NomicFoundation/edr/pull/671/commits/9d3adad2a37095385ca9a6f7ff5c9c719e842f2c

**Pull request merged**